### ISSUE TITLE
Knuckles Ledge Jump Animation

### ISSUE THREAD

**Issue by Yohan1044** [Project-M-CC/knuckles#3]
While a perfect Knuckles might have decloned auxiliary and item animations from Sonic, this is a lot of work. But as it stands, Knuckles Ledge Jump animation lacks the loading of the sphere model... which causes Knuckles to flash in and out of existence when doing a jump from ledge.

**Comment by TheSpaceKing12** [comment]
https://www.youtube.com/watch?v=f_rmzAx3ZMM

Video for documentation. Don't really know if I showed the issue or not. If I did...well yay for me.

Thought it'd mention i'm recording these off wii with a dazzle capture card in S-video more.

**Comment by Yohan1044** [comment]
I didn't actually see the issue in the video... but it's very prevalent the last time I played with Knuckles.
For the record... PM has a built in debug mode (but only for GC controllers).

If you press L-R-and down taunt ... the start button becomes a debug-mode toggle instead of a pause/select button.
Hit start to enter debug mode, start again to stop it. L-R-downTaunt while not in debug mode to restore start to normal functionality.
While in debug mode... pressing Z will move the game forward by one frame. This lets you count frames and really slow things down.
While in debug mode... pressing R- and right taunt will toggle hitboxes being visible.
There's more... but those are the most handy features in my opinion.

In my game, Knuckles very much disapeers completely when jumping from the ledge.

**Comment by TheSpaceKing12** [comment]
Perhaps it's just an issue with your specific build your playing on? I'll try it again with debug enabled. And will report back my findings either way.

i'm a fresh spring chicken when it comes to modding. Like first couple weeks new. lol. I figured I could at the very least out by filming reported issues in action. For documentation and organization's sake.

**Comment by TheSpaceKing12** [comment]
https://youtu.be/_9r8L-K7kxI

Take two. Do you see it Yohan? It looks freaky. lol

I will say though. Watching those animations frame by frame. It's ugly as hell. xD

**Comment by Yohan1044** [comment]
Nope... that looks good. It could just be the version of Knuckles, my instance of Dolphin and the way it's configured, or something else. I don't see his sphere model loading... but only during these ledge jumps. I'll look into it eventually.

I like these videos. As for modding experience, it's just something you have to jump right into. I hope knowledge of debug mode helps you in days to come. Be warned though... PSA rigs hitboxes to bones based off the costume... but debug mode riggs hitboxes to bones based off the character. If using a clone with a new bone structure... you'll see hitboxes in places they really aren't.
